src/contracts/core/DelegationManager.sol

```diff
@@ -528,131 +633,224 @@ contract DelegationManager is
-                    address podOwnerOperator = delegatedTo[staker];
-                    // Similar to `isDelegated` logic
-                    if (podOwnerOperator != address(0)) {
-                        _increaseOperatorShares({
-                            operator: podOwnerOperator,
-                            // the 'staker' here is the address receiving new shares
-                            staker: staker,
-                            strategy: withdrawal.strategies[i],
-                            shares: increaseInDelegateableShares
-                        });
-                    }
-                } else {
-                    strategyManager.addShares(msg.sender, tokens[i], withdrawal.strategies[i], withdrawal.shares[i]);
-                    // Similar to `isDelegated` logic
-                    if (currentOperator != address(0)) {
-                        _increaseOperatorShares({
-                            operator: currentOperator,
-                            // the 'staker' here is the address receiving new shares
-                            staker: msg.sender,
-                            strategy: withdrawal.strategies[i],
-                            shares: withdrawal.shares[i]
-                        });
-                    }
+                (shares, existingShares) = eigenPodManager.addShares({podOwner: staker, shares: shares});
+                address podOwnerOperator = delegatedTo[staker];
+                // Similar to `isDelegated` logic
+                if (podOwnerOperator != address(0)) {
+                    _increaseOperatorScaledShares({
+                        operator: podOwnerOperator,
+                        // the 'staker' here is the address receiving new shares
+                        staker: staker,
+                        strategy: withdrawal.strategies[i],
+                        shares: shares,
+                        totalMagnitude: totalMagnitudes[i]
+                    });
                 }
-                unchecked {
-                    ++i;
+            } else {
+                existingShares = strategyManager.addShares(msg.sender, tokens[i], withdrawal.strategies[i], shares);
+                // Similar to `isDelegated` logic
+                if (currentOperator != address(0)) {
+                    _increaseOperatorScaledShares({
+                        operator: currentOperator,
+                        // the 'staker' here is the address receiving new shares
+                        staker: msg.sender,
+                        strategy: withdrawal.strategies[i],
+                        shares: shares,
+                        totalMagnitude: totalMagnitudes[i]
+                    });
                 }
             }
-        }
 
-        emit WithdrawalCompleted(withdrawalRoot);
+            // update stakers scaling deposit scaling factor
+            uint256 newStakerScalingFactor = _calculateStakerScalingFactor({
```

Comment:
so i think we should instead have the `addShares` call back into the DM just like a regular deposit